extensions/2.0/Vendor/KITTYCAD_boundary_representation/README.md

```diff
@@ -0,0 +1,250 @@
+
+## Concepts
+
+### Solid
+
+A _solid_ represents a single volume of solid material. The boundary of the volume is called the solid's _outer shell_. In addition to its outer shell, a solid may have internal hollow regions; such regions are called _inner shells_. Both outer shells and inner shells are represented by the `Shell` data structure.
+
+![solid-example](figures/solid.png)
+
+#### Example
+
+The interior hollow space of a tennis ball would be represented by an inner shell. The surface of the tennis ball would be represented by its outer shell.
+
+![tennis-ball-cross-section](https://content.instructables.com/FRV/UI1L/J2AW02WE/FRVUI1LJ2AW02WE.jpg?auto=webp&frame=1&width=1024&height=1024&fit=bounds&md=da5cf8a96b88acbea9eb082a7b378ae7)
+
+### Shell
+
+A _shell_ is a collection of _faces_ in 3D space which form a 'watertight' volume. A shell is represented by the `Shell` data structure.
+
+### Face
+
+A _face_ is a portion of a _surface_ in 3D space. Multiple faces are referenced to form _shells_.
+
+Faces are associated with the 3D surface they are defined on. The boundary of faces are defined by a set of _edges_ which form a closed _loop_ in 3D space, following the contours of its associated surface. The boundary of a face is defined by its _outer loop_. In a similar manner to _shells_, faces may have internal hollow regions; such regions are called _inner loops_. Both outer loops and inner loops are represented by the `Loop` data structure.
+
+![face-example](figures/face.png)
+
+#### Example
+
+This face is formed by a closed loop of 3D curves which are coincident to its associated planar surface. The grid visualizes the associated planar surface.
+
+### Loop
+
+A _loop_ is a set of _edges_ that combine to form a closed circuit. Edges are referenced in the loop so that their direction may be reversed.
+
+![loop-example](figures/loop.png)
+
+### Curve
+
+A _curve_ defines a function that evaluates a single parameter to a point in either 2D or 3D space, depending on context.
+
+Curves are referenced by _edges_ to define the boundary between _faces_ in 3D space. Supplementary curves may be referenced by _loops_ to define the same boundary between _faces_ in 2D _surface_ parameter space.
+
+#### Base structure
+
+```json
+{
+  "type": "curveType",
+  "curveType": {
+    "curveSpecificValue": 1
+  },
+  "domain": {
+    "min": 0,
+    "max": 1
+  }
+}
+```
+
+#### Line (start & direction)
+
+```json
+"line": {
+  "start": [-1, -1, -1],
+  "direction": [1, 1, 1],
+}
+```
+
+#### Line (start & end)
+
+
+```json
+"line": {
+  "start": [-1, -1, -1],
+  "end": [1, 1, 1],
+}
+```
+
+#### Circle
```

Comment:
Would it be more flexible for a curve to be "circular arc" rather than a "circle"?

Reply:
It's a good point. I think this is a matter of naming. All curves have an optional subdomain so they are meant to be flexible in this way already, i.e., a circle curve with a subdomain would be an arc of the circle.